gqcp/tests/Operator/SecondQuantized/USQTwoElectronOperator_test.cpp

```diff
@@ -0,0 +1,452 @@
+
+}
+
+
+/**
+ *  Check if the zero constructor really sets is parameters to all zeros
+ */
+BOOST_AUTO_TEST_CASE ( USQTwoElectronOperator_zero_constructor ) {
+
+    const size_t K = 2;
+    GQCP::ScalarUSQTwoElectronOperator<double> op {K}; // should initialize zero's
+
+    // Create a reference zero tensor
+    GQCP::QCRankFourTensor<double> ref (K);
+
+    for (size_t i = 0; i < K; i++) {
+        for (size_t j = 0; j < K; j++) {
+            for (size_t k = 0; k < K; k++) {
+                for (size_t l = 0; l < K; l++) {
+                    ref(i,j,k,l) = 0;
+                }
+            }
+        }
+    }
+
+    BOOST_CHECK_EQUAL(op.alphaAlphaDimension(), K);
+    BOOST_CHECK_EQUAL(op.alphaBetaDimension(), K);
+    BOOST_CHECK_EQUAL(op.betaAlphaDimension(), K);
+    BOOST_CHECK_EQUAL(op.betaBetaDimension(), K);
+
+    BOOST_CHECK(op.alphaAlphaParameters().isApprox(ref, 1.0e-08));
+    BOOST_CHECK(op.alphaBetaParameters().isApprox(ref, 1.0e-08));
+    BOOST_CHECK(op.betaAlphaParameters().isApprox(ref, 1.0e-08));
+    BOOST_CHECK(op.betaBetaParameters().isApprox(ref, 1.0e-08));
+
+}
+
+
+/**
+ *  Check if calculateExpectationValue throws when necessary
+ */
+BOOST_AUTO_TEST_CASE ( calculateExpectationValue_throw ) {
+
+    const GQCP::ScalarUSQTwoElectronOperator<double> g {2};
+
+    const GQCP::TwoRDM<double> D_valid (2);
+    const GQCP::TwoRDM<double> D_invalid (3);
+
+    BOOST_CHECK_THROW(g.calculateExpectationValue(D_invalid, D_invalid, D_invalid, D_invalid), std::invalid_argument);
+    BOOST_CHECK_THROW(g.calculateExpectationValue(D_invalid, D_valid, D_invalid, D_invalid), std::invalid_argument);
+    BOOST_CHECK_THROW(g.calculateExpectationValue(D_valid, D_invalid, D_valid, D_invalid), std::invalid_argument);
+    BOOST_CHECK_THROW(g.calculateExpectationValue(D_valid, D_invalid, D_valid, D_valid), std::invalid_argument);
+
+    BOOST_CHECK_NO_THROW(g.calculateExpectationValue(D_valid, D_valid,D_valid, D_valid));
+}
+
+
+/**
+ * Check whether or not calculateExpectationValue shows the correct behaviour
+ */
+BOOST_AUTO_TEST_CASE ( calculateExpectationValue_behaviour ) {
+    
+    const size_t dim = 2;
+
+   // Initialize a test tensor and convert it into an operators
+    GQCP::QCRankFourTensor<double> T1 (dim);
+
+    for (size_t i = 0; i < dim; i++) {
+        for (size_t j = 0; j < dim; j++) {
+            for (size_t k = 0; k < dim; k++) {
+                for (size_t l = 0; l < dim; l++) {
+                    T1(i,j,k,l) = (i+1) + 2*(j+1) + 4*(k+1) + 8*(l+1);
+                }
+            }
+        }
+    }
+
+    GQCP::QCRankFourTensor<double> T2 (dim);
+
+    for (size_t i = 0; i < dim; i++) {
+        for (size_t j = 0; j < dim; j++) {
+            for (size_t k = 0; k < dim; k++) {
+                for (size_t l = 0; l < dim; l++) {
+                    T2(i,j,k,l) = 0;
+                }
+            }
+        }
+    }
```

Comment:
I think it would be better for testing purposes if you actually used a non-zero `T2` here.